repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class RepairOrder(models.Model):
+    """Extend repair.order with grouping, cascade, and shared quotation logic.
+
+    Features added:
+      • Link repairs into logical groups (via `repair.order.group`)
+      • Automatically create groups when adding another repair
+      • Synchronize customer (partner) across all repairs in the same group
+      • Cascade confirmation/cancellation actions within the group
+      • Create a single sale order for all repairs of one group
+    """
+
+    _inherit = "repair.order"
+
+    group_id = fields.Many2one(
+        "repair.order.group",
+        string="Repair Group",
+        ondelete="set null",
+        check_company=True,
+        index=True,
+        help="Repairs sharing the same group can be processed and quoted together.",
+    )
+
+    grouped_repair_ids = fields.One2many(
```

Comment:
You can make this field computed and store only orders other than the current one. This will simplify the code because there will be no need to check for siblings any longer.

```python
grouped_repair_ids = fields.One2many(compute="_compute_grouped_repair_ids")
...

@api.depends("group_id", "group_id.repair_ids")
def _compute_grouped_repair_ids(self):
    for repair in self:
        if not repair.group_id:
             repair.group_repair_ids = False
             continue
        repair.group_repair_ids = repair.group_id.repair_ids - repair
```